commons-numbers-arrays/src/test/java/org/apache/commons/numbers/arrays/ExtendedPrecisionTest.java

```diff
@@ -102,6 +102,23 @@ void testSubNormalSplit() {
         final double lo2 = a - hi2;
         Assertions.assertEquals(a, hi2);
         Assertions.assertEquals(0, lo2);
+
+
         Assertions.assertTrue(Math.abs(hi2) > Math.abs(lo2));
     }
+
+    @Test
+    void testSquareLowUnscaled() {
+        assertSquareLowUnscaled(0.0, 1.0);
+        assertSquareLowUnscaled(0.0, -1.0);
+        assertSquareLowUnscaled(-1.4293872661474477E-16, Math.PI);
```

Comment:
Can you compute this result using BigDecimal?
```java
final BigDecimal pi = new BigDecimal(Math.PI);
final double pi2low = pi.pow(2).subtract(new BigDecimal(Math.PI * Math.PI)).doubleValue();
assertSquareLowUnscaled(pi2low, Math.PI);
```

It is a bit cryptic to have a constant here.

Reply:
That sounds good but the expected value computed with `BigDecimal` is `-8.79470486217151E-16` whereas the `ExtendedPrecision` result is '-1.4293872661474477E-16'. Is this simply a limitation of the algorithm or is this a bug in our implementation?

Reply:
I thought the BigDecimal value was the same. I just tried the code above on a different machine and it still works. Stepping through a debugger the value of `pi2low` is `-1.429...`. Are you using a rounding mode for BigDecimal?

Reply:
Since the code is on Java 9 for testing you can drop BigDecimal and simplify this to:
```java
assertSquareLowUnscaled(Math.fma(Math.PI, Math.PI, -Math.PI * Math.PI), Math.PI);
```

Reply:
I found why `BigDecimal` wasn't working for me: I was using `BigDecimal.valueOf()` to create instances instead of `new BigDecimal()`. Switching to the constructor to use the exact double representation fixed the issue. I'm switching to the `fma` version, however since that is simpler.